### PR TITLE
Allow `-` and `_` for source ids.

### PR DESCRIPTION
**Pull request by motizuki**

### CODE REVIEW

File: app/models/source.rb [hunk 1]
```diff
@@ -40,6 +40,7 @@ def create_link_check_rule
   end
 
   def slugify_source_id
-    self.source_id = source_id.parameterize.underscore
+    # Convert any special characters to underscore except hiphens
```
Comment (tim-boost) [inline app/models/source.rb]: small thing, should be 'hyphens'

File: spec/models/source_spec.rb [hunk 2]
```diff
@@ -80,6 +80,12 @@
       source.save!
       expect(source.source_id).to eq '84_ource'
     end
+
+    it 'retain - and _ from source_id' do
```
Comment (theomarkkuspaul) [inline spec/models/source_spec.rb]: `'retains'`

File: app/models/source.rb (hunk 1 above)
Comment (motizuki) [reply app/models/source.rb]: 👍

File: spec/models/source_spec.rb (hunk 2 above)
Comment (motizuki) [reply spec/models/source_spec.rb]: 👍

**Pull request merged**